// ngx-cookieconsent/esm2020/lib/ngx-cookieconsent.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { WindowService, NgcCookieConsentConfig, NgcCookieConsentService } from './service/index';
import * as i0 from "@angular/core";
/**
 * Main module of the library
 */
export class NgcCookieConsentModule {
    static forRoot(config) {
        return {
            ngModule: NgcCookieConsentModule,
            providers: [WindowService, { provide: NgcCookieConsentConfig, useValue: config }, NgcCookieConsentService]
        };
    }
}
NgcCookieConsentModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.5", ngImport: i0, type: NgcCookieConsentModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
NgcCookieConsentModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "14.2.5", ngImport: i0, type: NgcCookieConsentModule, imports: [CommonModule] });
NgcCookieConsentModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "14.2.5", ngImport: i0, type: NgcCookieConsentModule, imports: [CommonModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.5", ngImport: i0, type: NgcCookieConsentModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule
                    ]
                }]
        }] });
