// ngx-qrcode-styling/esm2022/lib/ngx-qrcode-styling.service.mjs
import { Injectable } from '@angular/core';
import { AsyncSubject } from 'rxjs';
import { deepUpdate, defaultTemplate, drawQrcode } from './ngx-qrcode-styling.helper';
import * as i0 from "@angular/core";
class NgxQrcodeStylingService {
    /**
     * create
     * @param config
     * @param container
     * @returns
     */
    create(config, container) {
        return drawQrcode(defaultTemplate(config), container);
    }
    /**
     * update
     * @param config
     * @param configUpdate
     * @param container
     * @returns
     */
    update(config, configUpdate, container) {
        const subject = new AsyncSubject();
        (async function () {
            const conf = await deepUpdate(defaultTemplate(config), defaultTemplate(configUpdate));
            drawQrcode(conf, container).subscribe(s => {
                subject.next(s);
                subject.complete();
            });
        })();
        return subject;
    }
    /**
     * download image
     * @param fileName eg: demo.png
     * @param container
     * @param timeout
     * @returns
     */
    download(container, fileName = '', timeout = 0) {
        const subject = new AsyncSubject();
        const _fileName = (type) => {
            if (!fileName) {
                fileName = `ngx_qrcode_styling_${Date.now()}.${type}`;
            }
        };
        const _download = (dataURL) => {
            const link = document.createElement('a');
            link.download = fileName;
            link.href = dataURL;
            link.click();
            link.remove();
        };
        const _complete = (url) => {
            if (url) {
                _download(url);
                subject.next({ fileName, container, url: url });
                subject.complete();
            }
            else {
                subject.error('Container not found!');
                subject.complete();
            }
        };
        setTimeout(() => {
            const canvas = container?.querySelector('canvas');
            if (canvas) {
                _fileName('png');
                const typeImg = fileName ? fileName?.split('.')?.slice(-1)?.toString() : 'png';
                canvas.toBlob((blob) => _complete(URL.createObjectURL(blob)), 'image/' + typeImg);
            }
            else {
                const svg = container?.querySelector('svg');
                if (svg) {
                    _fileName('svg');
                    const svgData = svg.outerHTML;
                    const svgBlob = new Blob([svgData], { type: "image/svg+xml;charset=utf-8" });
                    const svgUrl = URL.createObjectURL(svgBlob);
                    _complete(svgUrl);
                }
                else {
                    _complete('');
                }
            }
        }, timeout);
        return subject;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.0.4", ngImport: i0, type: NgxQrcodeStylingService, deps: [], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "16.0.4", ngImport: i0, type: NgxQrcodeStylingService, providedIn: 'root' }); }
}
export { NgxQrcodeStylingService };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.0.4", ngImport: i0, type: NgxQrcodeStylingService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }] });
