// ngx-qrcode-styling/esm2022/lib/ngx-qrcode-styling.module.mjs
import { NgModule } from '@angular/core';
import { NgxQrcodeStylingComponent } from './ngx-qrcode-styling.component';
import { NgxQrcodeStylingService } from './ngx-qrcode-styling.service';
import * as i0 from "@angular/core";
class NgxQrcodeStylingModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.0.4", ngImport: i0, type: NgxQrcodeStylingModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.0.4", ngImport: i0, type: NgxQrcodeStylingModule, declarations: [NgxQrcodeStylingComponent], exports: [NgxQrcodeStylingComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.0.4", ngImport: i0, type: NgxQrcodeStylingModule, providers: [
            NgxQrcodeStylingService
        ] }); }
}
export { NgxQrcodeStylingModule };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.0.4", ngImport: i0, type: NgxQrcodeStylingModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        NgxQrcodeStylingComponent
                    ],
                    exports: [
                        NgxQrcodeStylingComponent
                    ],
                    providers: [
                        NgxQrcodeStylingService
                    ]
                }]
        }] });
