// @vime/angular/esm2015/components/ControlGroup.js
import { __decorate } from "tslib";
/* eslint-disable */
/* tslint:disable */
// @ts-nocheck
import { ChangeDetectionStrategy, ChangeDetectorRef, Component, ElementRef, NgZone } from '@angular/core';
import { ProxyCmp, initOutputs } from '../lib';
import { VmControlGroup } from '@vime/core';
import { define } from '../lib';
const ControlGroupInputs = [
    'space',
];
const ControlGroupMethods = [];
const ControlGroupOutputs = [];
let ControlGroup = class ControlGroup {
    constructor(c, r, z) {
        this.z = z;
        define('vm-control-group', VmControlGroup);
        c.detach();
        this.el = r.nativeElement;
        initOutputs(this, []);
    }
};
ControlGroup.decorators = [
    { type: Component, args: [{
                selector: 'vm-control-group',
                changeDetection: ChangeDetectionStrategy.OnPush,
                template: '<ng-content></ng-content>',
                inputs: ControlGroupInputs,
                outputs: ControlGroupOutputs
            },] }
];
ControlGroup.ctorParameters = () => [
    { type: ChangeDetectorRef },
    { type: ElementRef },
    { type: NgZone }
];
ControlGroup = __decorate([
    ProxyCmp({
        inputs: ControlGroupInputs,
        methods: ControlGroupMethods,
    })
], ControlGroup);
export { ControlGroup };
