// @vime/angular/esm2015/components/Slider.js
import { __decorate } from "tslib";
/* eslint-disable */
/* tslint:disable */
// @ts-nocheck
import { ChangeDetectionStrategy, ChangeDetectorRef, Component, ElementRef, NgZone } from '@angular/core';
import { ProxyCmp, initOutputs } from '../lib';
import { VmSlider } from '@vime/core';
import { define } from '../lib';
const SliderInputs = [
    'step',
    'min',
    'max',
    'value',
    'valueText',
    'label',
];
const SliderMethods = [];
const SliderOutputs = [
    'vmValueChange',
    'vmFocus',
    'vmBlur',
];
let Slider = class Slider {
    constructor(c, r, z) {
        this.z = z;
        define('vm-slider', VmSlider);
        c.detach();
        this.el = r.nativeElement;
        initOutputs(this, ['vmValueChange', 'vmFocus', 'vmBlur']);
    }
};
Slider.decorators = [
    { type: Component, args: [{
                selector: 'vm-slider',
                changeDetection: ChangeDetectionStrategy.OnPush,
                template: '<ng-content></ng-content>',
                inputs: SliderInputs,
                outputs: SliderOutputs
            },] }
];
Slider.ctorParameters = () => [
    { type: ChangeDetectorRef },
    { type: ElementRef },
    { type: NgZone }
];
Slider = __decorate([
    ProxyCmp({
        inputs: SliderInputs,
        methods: SliderMethods,
    })
], Slider);
export { Slider };
