// @vime/angular/esm2015/components/Spinner.js
import { __decorate } from "tslib";
/* eslint-disable */
/* tslint:disable */
// @ts-nocheck
import { ChangeDetectionStrategy, ChangeDetectorRef, Component, ElementRef, NgZone } from '@angular/core';
import { ProxyCmp, initOutputs } from '../lib';
import { VmSpinner } from '@vime/core';
import { define } from '../lib';
const SpinnerInputs = [
    'isVideoView',
    'currentProvider',
    'showWhenMediaLoading',
    'playbackReady',
    'buffering',
];
const SpinnerMethods = [];
const SpinnerOutputs = [
    'vmWillShow',
    'vmWillHide',
];
let Spinner = class Spinner {
    constructor(c, r, z) {
        this.z = z;
        define('vm-spinner', VmSpinner);
        c.detach();
        this.el = r.nativeElement;
        initOutputs(this, ['vmWillShow', 'vmWillHide']);
    }
};
Spinner.decorators = [
    { type: Component, args: [{
                selector: 'vm-spinner',
                changeDetection: ChangeDetectionStrategy.OnPush,
                template: '<ng-content></ng-content>',
                inputs: SpinnerInputs,
                outputs: SpinnerOutputs
            },] }
];
Spinner.ctorParameters = () => [
    { type: ChangeDetectorRef },
    { type: ElementRef },
    { type: NgZone }
];
Spinner = __decorate([
    ProxyCmp({
        inputs: SpinnerInputs,
        methods: SpinnerMethods,
    })
], Spinner);
export { Spinner };
