// @vime/angular/esm2015/components/Submenu.js
import { __decorate } from "tslib";
/* eslint-disable */
/* tslint:disable */
// @ts-nocheck
import { ChangeDetectionStrategy, ChangeDetectorRef, Component, ElementRef, NgZone } from '@angular/core';
import { ProxyCmp, initOutputs } from '../lib';
import { VmSubmenu, VmIcon, VmMenu, VmMenuItem } from '@vime/core';
import { define } from '../lib';
const SubmenuInputs = [
    'label',
    'hint',
    'slideInDirection',
    'active',
];
const SubmenuMethods = [
    'getController',
    'getMenu',
    'getControllerHeight',
];
const SubmenuOutputs = [
    'vmOpenSubmenu',
    'vmCloseSubmenu',
];
let Submenu = class Submenu {
    constructor(c, r, z) {
        this.z = z;
        define('vm-submenu', VmSubmenu);
        define('vm-icon', VmIcon);
        define('vm-menu', VmMenu);
        define('vm-menu-item', VmMenuItem);
        c.detach();
        this.el = r.nativeElement;
        initOutputs(this, ['vmOpenSubmenu', 'vmCloseSubmenu']);
    }
};
Submenu.decorators = [
    { type: Component, args: [{
                selector: 'vm-submenu',
                changeDetection: ChangeDetectionStrategy.OnPush,
                template: '<ng-content></ng-content>',
                inputs: SubmenuInputs,
                outputs: SubmenuOutputs
            },] }
];
Submenu.ctorParameters = () => [
    { type: ChangeDetectorRef },
    { type: ElementRef },
    { type: NgZone }
];
Submenu = __decorate([
    ProxyCmp({
        inputs: SubmenuInputs,
        methods: SubmenuMethods,
    })
], Submenu);
export { Submenu };
