// @vime/angular/esm2015/components/TimeProgress.js
import { __decorate } from "tslib";
/* eslint-disable */
/* tslint:disable */
// @ts-nocheck
import { ChangeDetectionStrategy, ChangeDetectorRef, Component, ElementRef, NgZone } from '@angular/core';
import { ProxyCmp, initOutputs } from '../lib';
import { VmTimeProgress, VmCurrentTime, VmTime, VmEndTime } from '@vime/core';
import { define } from '../lib';
const TimeProgressInputs = [
    'separator',
    'alwaysShowHours',
];
const TimeProgressMethods = [];
const TimeProgressOutputs = [];
let TimeProgress = class TimeProgress {
    constructor(c, r, z) {
        this.z = z;
        define('vm-time-progress', VmTimeProgress);
        define('vm-current-time', VmCurrentTime);
        define('vm-time', VmTime);
        define('vm-end-time', VmEndTime);
        c.detach();
        this.el = r.nativeElement;
        initOutputs(this, []);
    }
};
TimeProgress.decorators = [
    { type: Component, args: [{
                selector: 'vm-time-progress',
                changeDetection: ChangeDetectionStrategy.OnPush,
                template: '<ng-content></ng-content>',
                inputs: TimeProgressInputs,
                outputs: TimeProgressOutputs
            },] }
];
TimeProgress.ctorParameters = () => [
    { type: ChangeDetectorRef },
    { type: ElementRef },
    { type: NgZone }
];
TimeProgress = __decorate([
    ProxyCmp({
        inputs: TimeProgressInputs,
        methods: TimeProgressMethods,
    })
], TimeProgress);
export { TimeProgress };
