// @vime/angular/esm2015/components/Time.js
import { __decorate } from "tslib";
/* eslint-disable */
/* tslint:disable */
// @ts-nocheck
import { ChangeDetectionStrategy, ChangeDetectorRef, Component, ElementRef, NgZone } from '@angular/core';
import { ProxyCmp, initOutputs } from '../lib';
import { VmTime } from '@vime/core';
import { define } from '../lib';
const TimeInputs = [
    'label',
    'seconds',
    'alwaysShowHours',
];
const TimeMethods = [];
const TimeOutputs = [];
let Time = class Time {
    constructor(c, r, z) {
        this.z = z;
        define('vm-time', VmTime);
        c.detach();
        this.el = r.nativeElement;
        initOutputs(this, []);
    }
};
Time.decorators = [
    { type: Component, args: [{
                selector: 'vm-time',
                changeDetection: ChangeDetectionStrategy.OnPush,
                template: '<ng-content></ng-content>',
                inputs: TimeInputs,
                outputs: TimeOutputs
            },] }
];
Time.ctorParameters = () => [
    { type: ChangeDetectorRef },
    { type: ElementRef },
    { type: NgZone }
];
Time = __decorate([
    ProxyCmp({
        inputs: TimeInputs,
        methods: TimeMethods,
    })
], Time);
export { Time };
