// @vime/angular/esm2015/components/Tooltip.js
import { __decorate } from "tslib";
/* eslint-disable */
/* tslint:disable */
// @ts-nocheck
import { ChangeDetectionStrategy, ChangeDetectorRef, Component, ElementRef, NgZone } from '@angular/core';
import { ProxyCmp, initOutputs } from '../lib';
import { VmTooltip } from '@vime/core';
import { define } from '../lib';
const TooltipInputs = [
    'hidden',
    'active',
    'position',
    'direction',
    'isTouch',
    'isMobile',
];
const TooltipMethods = [];
const TooltipOutputs = [];
let Tooltip = class Tooltip {
    constructor(c, r, z) {
        this.z = z;
        define('vm-tooltip', VmTooltip);
        c.detach();
        this.el = r.nativeElement;
        initOutputs(this, []);
    }
};
Tooltip.decorators = [
    { type: Component, args: [{
                selector: 'vm-tooltip',
                changeDetection: ChangeDetectionStrategy.OnPush,
                template: '<ng-content></ng-content>',
                inputs: TooltipInputs,
                outputs: TooltipOutputs
            },] }
];
Tooltip.ctorParameters = () => [
    { type: ChangeDetectorRef },
    { type: ElementRef },
    { type: NgZone }
];
Tooltip = __decorate([
    ProxyCmp({
        inputs: TooltipInputs,
        methods: TooltipMethods,
    })
], Tooltip);
export { Tooltip };
