// @vime/angular/esm2015/components/Ui.js
import { __decorate } from "tslib";
/* eslint-disable */
/* tslint:disable */
// @ts-nocheck
import { ChangeDetectionStrategy, ChangeDetectorRef, Component, ElementRef, NgZone } from '@angular/core';
import { ProxyCmp, initOutputs } from '../lib';
import { VmUi } from '@vime/core';
import { define } from '../lib';
const UiInputs = [
    'isVideoView',
    'playsinline',
    'isFullscreenActive',
];
const UiMethods = [];
const UiOutputs = [];
let Ui = class Ui {
    constructor(c, r, z) {
        this.z = z;
        define('vm-ui', VmUi);
        c.detach();
        this.el = r.nativeElement;
        initOutputs(this, []);
    }
};
Ui.decorators = [
    { type: Component, args: [{
                selector: 'vm-ui',
                changeDetection: ChangeDetectionStrategy.OnPush,
                template: '<ng-content></ng-content>',
                inputs: UiInputs,
                outputs: UiOutputs
            },] }
];
Ui.ctorParameters = () => [
    { type: ChangeDetectorRef },
    { type: ElementRef },
    { type: NgZone }
];
Ui = __decorate([
    ProxyCmp({
        inputs: UiInputs,
        methods: UiMethods,
    })
], Ui);
export { Ui };
