// @vime/angular/esm2015/components/Video.js
import { __decorate } from "tslib";
/* eslint-disable */
/* tslint:disable */
// @ts-nocheck
import { ChangeDetectionStrategy, ChangeDetectorRef, Component, ElementRef, NgZone } from '@angular/core';
import { ProxyCmp, initOutputs } from '../lib';
import { VmVideo, VmFile } from '@vime/core';
import { define } from '../lib';
const VideoInputs = [
    'willAttach',
    'hasCustomTextManager',
    'crossOrigin',
    'preload',
    'poster',
    'controlsList',
    'autoPiP',
    'disablePiP',
    'disableRemotePlayback',
    'mediaTitle',
];
const VideoMethods = [
    'getAdapter',
];
const VideoOutputs = [];
let Video = class Video {
    constructor(c, r, z) {
        this.z = z;
        define('vm-video', VmVideo);
        define('vm-file', VmFile);
        c.detach();
        this.el = r.nativeElement;
        initOutputs(this, []);
    }
};
Video.decorators = [
    { type: Component, args: [{
                selector: 'vm-video',
                changeDetection: ChangeDetectionStrategy.OnPush,
                template: '<ng-content></ng-content>',
                inputs: VideoInputs,
                outputs: VideoOutputs
            },] }
];
Video.ctorParameters = () => [
    { type: ChangeDetectorRef },
    { type: ElementRef },
    { type: NgZone }
];
Video = __decorate([
    ProxyCmp({
        inputs: VideoInputs,
        methods: VideoMethods,
    })
], Video);
export { Video };
