// @vime/angular/esm2015/components/Vimeo.js
import { __decorate } from "tslib";
/* eslint-disable */
/* tslint:disable */
// @ts-nocheck
import { ChangeDetectionStrategy, ChangeDetectorRef, Component, ElementRef, NgZone } from '@angular/core';
import { ProxyCmp, initOutputs } from '../lib';
import { VmVimeo, VmEmbed } from '@vime/core';
import { define } from '../lib';
const VimeoInputs = [
    'videoId',
    'byline',
    'color',
    'portrait',
    'noAutoAspectRatio',
    'poster',
    'cookies',
    'language',
    'aspectRatio',
    'autoplay',
    'controls',
    'logger',
    'loop',
    'muted',
    'playsinline',
];
const VimeoMethods = [
    'getAdapter',
];
const VimeoOutputs = [
    'vmError',
];
let Vimeo = class Vimeo {
    constructor(c, r, z) {
        this.z = z;
        define('vm-vimeo', VmVimeo);
        define('vm-embed', VmEmbed);
        c.detach();
        this.el = r.nativeElement;
        initOutputs(this, ['vmError']);
    }
};
Vimeo.decorators = [
    { type: Component, args: [{
                selector: 'vm-vimeo',
                changeDetection: ChangeDetectionStrategy.OnPush,
                template: '<ng-content></ng-content>',
                inputs: VimeoInputs,
                outputs: VimeoOutputs
            },] }
];
Vimeo.ctorParameters = () => [
    { type: ChangeDetectorRef },
    { type: ElementRef },
    { type: NgZone }
];
Vimeo = __decorate([
    ProxyCmp({
        inputs: VimeoInputs,
        methods: VimeoMethods,
    })
], Vimeo);
export { Vimeo };
