// @vime/angular/esm2015/components/VolumeControl.js
import { __decorate } from "tslib";
/* eslint-disable */
/* tslint:disable */
// @ts-nocheck
import { ChangeDetectionStrategy, ChangeDetectorRef, Component, ElementRef, NgZone } from '@angular/core';
import { ProxyCmp, initOutputs } from '../lib';
import { VmVolumeControl, VmControl, VmIcon, VmMuteControl, VmTooltip, VmSlider } from '@vime/core';
import { define } from '../lib';
const VolumeControlInputs = [
    'lowVolumeIcon',
    'highVolumeIcon',
    'mutedIcon',
    'icons',
    'tooltipPosition',
    'tooltipDirection',
    'hideTooltip',
    'muteKeys',
    'noKeyboard',
    'muted',
    'volume',
    'isMobile',
    'i18n',
];
const VolumeControlMethods = [];
const VolumeControlOutputs = [];
let VolumeControl = class VolumeControl {
    constructor(c, r, z) {
        this.z = z;
        define('vm-volume-control', VmVolumeControl);
        define('vm-control', VmControl);
        define('vm-icon', VmIcon);
        define('vm-mute-control', VmMuteControl);
        define('vm-tooltip', VmTooltip);
        define('vm-slider', VmSlider);
        c.detach();
        this.el = r.nativeElement;
        initOutputs(this, []);
    }
};
VolumeControl.decorators = [
    { type: Component, args: [{
                selector: 'vm-volume-control',
                changeDetection: ChangeDetectionStrategy.OnPush,
                template: '<ng-content></ng-content>',
                inputs: VolumeControlInputs,
                outputs: VolumeControlOutputs
            },] }
];
VolumeControl.ctorParameters = () => [
    { type: ChangeDetectorRef },
    { type: ElementRef },
    { type: NgZone }
];
VolumeControl = __decorate([
    ProxyCmp({
        inputs: VolumeControlInputs,
        methods: VolumeControlMethods,
    })
], VolumeControl);
export { VolumeControl };
