// @vime/angular/esm2015/components/Youtube.js
import { __decorate } from "tslib";
/* eslint-disable */
/* tslint:disable */
// @ts-nocheck
import { ChangeDetectionStrategy, ChangeDetectorRef, Component, ElementRef, NgZone } from '@angular/core';
import { ProxyCmp, initOutputs } from '../lib';
import { VmYoutube, VmEmbed } from '@vime/core';
import { define } from '../lib';
const YoutubeInputs = [
    'cookies',
    'videoId',
    'showFullscreenControl',
    'poster',
    'language',
    'autoplay',
    'controls',
    'logger',
    'loop',
    'muted',
    'playsinline',
];
const YoutubeMethods = [
    'getAdapter',
];
const YoutubeOutputs = [];
let Youtube = class Youtube {
    constructor(c, r, z) {
        this.z = z;
        define('vm-youtube', VmYoutube);
        define('vm-embed', VmEmbed);
        c.detach();
        this.el = r.nativeElement;
        initOutputs(this, []);
    }
};
Youtube.decorators = [
    { type: Component, args: [{
                selector: 'vm-youtube',
                changeDetection: ChangeDetectionStrategy.OnPush,
                template: '<ng-content></ng-content>',
                inputs: YoutubeInputs,
                outputs: YoutubeOutputs
            },] }
];
Youtube.ctorParameters = () => [
    { type: ChangeDetectorRef },
    { type: ElementRef },
    { type: NgZone }
];
Youtube = __decorate([
    ProxyCmp({
        inputs: YoutubeInputs,
        methods: YoutubeMethods,
    })
], Youtube);
export { Youtube };
